code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -67,6 +67,9 @@
 		var/dmg = armor_damage_reduction(GLOB.xeno_fire, PASSIVE_BURN_DAM_CALC(fire_reagent.intensityfire, fire_reagent.durationfire, fire_stacks))
 		apply_damage(dmg, BURN)
 		INVOKE_ASYNC(src, /mob.proc/emote, pick("roar", "needhelp"))
+	if(FIRE_VULNERABILITY)
```

Comment:
This will always be true. I figure you want to check this against the relevant bitfield instead

Reply:
Good catch, fixed.